sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingBufferedSender.cs

```diff
@@ -108,15 +108,21 @@ public class SearchIndexingBufferedSender<T> : IDisposable, IAsyncDisposable
         protected SearchIndexingBufferedSender() { }
 
         /// <summary>
-        /// Creates a new instance of the SearchIndexingBufferedSender.
+        /// Creates a new instance of <see cref="SearchIndexingBufferedSender{T}"/> that
+        /// can be used to index search documents with intelligent batching,
+        /// automatic flushing, and retries for failed indexing actions.
         /// </summary>
         /// <param name="searchClient">
         /// The SearchClient used to send requests to the service.
         /// </param>
         /// <param name="options">
-        /// Provides the configuration options for the sender.
+        /// The <see cref="SearchIndexingBufferedSenderOptions{T}"/> to
+        /// customize the sender's behavior.
         /// </param>
-        internal SearchIndexingBufferedSender(
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when the <paramref name="searchClient"/> is null.
+        /// </exception>
+        public SearchIndexingBufferedSender(
```

Comment:
Should this still be `public` if you're making it creatable from the `SearchClient`?

Reply:
...and if you're not calling the helper from tests anymore (at least, I don't see it here but maybe GitHub is hiding it), I'm puzzled as to why tests are now passing.

Reply:
Making it public is part of the requested change. This is being used from the samples.
The other part of removing the helper will be tackled in a different PR.

The helper is used heavily in the tests. That is the reason for the now small diff in BatchingTests.cs.